development/Dockerfile

```diff
@@ -1,18 +1,73 @@
 
-WORKDIR /source
+# Runtime argument and environment setup
+ARG NAUTOBOT_ROOT=/opt/nautobot
+
+ENV prometheus_multiproc_dir=/prom_cache
+ENV NAUTOBOT_ROOT ${NAUTOBOT_ROOT}
+
+# Don't need virtual environments in container, parallelized installer has inconsistent behavior
+RUN poetry config virtualenvs.create false \
+    && poetry config installer.parallel false
+
+# !!! USE CAUTION WHEN MODIFYING LINES ABOVE
+# -------------------------------------------------------------------------------------
+# Plugin-specific system build/test dependencies.
+# 
+# Example: LDAP requires `libldap2-dev` to be apt-installed before the Python package.
+# -------------------------------------------------------------------------------------
+# --> Start safe to modify section
 
-# Copy in only pyproject.toml/poetry.lock to help with caching this layer if no updates to dependencies
-COPY poetry.lock pyproject.toml /source/
-# --no-root declares not to install the project package since we're wanting to take advantage of caching dependency installation
-# and the project is copied in and installed after this step
-RUN poetry install --no-interaction --no-ansi --no-root
+# Uncomment the line below if you are apt-installing any package.
+# RUN apt update
+# RUN apt install libldap2-dev
 
-# Copy in the rest of the source code and install local Nautobot plugin
+# --> Stop safe to modify section
+# -------------------------------------------------------------------------------------
+# Install Nautobot Plugin
+# -------------------------------------------------------------------------------------
+# !!! USE CAUTION WHEN MODIFYING LINES BELOW
+
+# Copy in the source code
+WORKDIR /source
 COPY . /source
-RUN poetry install --no-interaction --no-ansi
 
-COPY development/nautobot_config.py /opt/nautobot/nautobot_config.py
+# Get container's installed Nautobot version as a forced constraint
+# NAUTOBOT_VER may be a branch name and not a published release therefor we need to get the installed version
+#  so pip can use it to recognize local constraints.
+RUN pip show nautobot | grep "^Version: " | sed -e 's/Version: /nautobot==/' > constraints.txt
+
+# Use Poetry to grab dev dependencies from the lock file
+# Can be improved in Poetry 1.2 which allows `poetry install --only dev`
+#
+# We can't use the entire freeze as it takes forever to resolve with rigidly fixed non-direct dependencies,
+#   especially those that are only direct to Nautobot but the container included versions slightly mismatch
+RUN poetry export -f requirements.txt --without-hashes --output poetry_freeze_base.txt
+RUN poetry export -f requirements.txt --dev --without-hashes --output poetry_freeze_all.txt
+RUN sort poetry_freeze_base.txt poetry_freeze_all.txt | uniq -u > poetry_freeze_dev.txt
+
+# Install all local project as editable, constrained on Nautobot version, to get any additional
+#   direct dependencies of the plugin
+RUN pip install -c constraints.txt -e .
```

Comment:
This fails currently because this project isn't setup to be installed as editable.

Reply:
Fixed by: https://github.com/nautobot/nautobot-plugin-data-validation-engine/pull/30/commits/de31beb266f2119f2c6ef10bc2a623401e7fc5d9